File: api/Shop/Model/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Shop.Model;

public class Product
{
    [Key] public uint ID { get; set; }
    [Required] public string Name { get; set; }
    [MinLength(3)] public string Description { get; set; }
    [Required] public float Price { get; set; }
    public bool Enabled { get; set; } = false;
    public int SortOrder { get; set; }
    [ForeignKey("Product")] public uint CategoryID { get; set; }
}

File: api/shop/model/product.go
package model

import "time"

type Product struct {
	ID          uint    `gorm:"primaryKey" json:"id"`
	Name        string  `json:"name"`
	Description string  `json:"description"`
	Price       float32 `json:"price"`
	Enabled     *bool   `json:"enabled"`
	SortOrder   uint    `json:"sortOrder"`
	ImageURL    string  `json:"imageURL"`
	CreatedAt   time.Time
	CategoryID  uint `json:"categoryID"`
}
